modules/packages/LinearAlgebra.chpl

```diff
@@ -2313,23 +2313,64 @@ module Sparse {
 
   /* Return a CSR matrix constructed from internal representation:
 
-    - ``shape``: bounding box dimensions
+    - ``parentDom``: parent domain for sparse matrix, bounding box for nonzeros indices
     - ``data``: non-zero element values
     - ``indices``: non-zero row pointers
     - ``indptr``: index pointers
 
   */
+  proc CSRMatrix(parentDom: domain(2), data: [?nnzDom] ?eltType, indices: [nnzDom], indptr: [?indDom])
+        where indDom.rank == 1 && nnzDom.rank == 1 {
+    var ADom = CSRDomain(parentDom, indices, indptr);
+    var A: [ADom] eltType;
+    A.data = data;
+    return A;
+  }
+
+  /* Return a CSR domain constructed from internal representation */
+  proc CSRDomain(parentDom: domain(2), indices: [?nnzDom], indptr: [?indDom])
+        where indDom.rank == 1 && nnzDom.rank == 1 {
+    const rowRange = parentDom.dim(0).low..parentDom.dim(0).high;
+    var ADom: sparse subdomain(parentDom) dmapped CS(sortedIndices=false);
+
+    ADom.startIdxDom = {rowRange.low..rowRange.high+1};
+    ADom.startIdx = indptr;
+    ADom._nnz = indices.size;
+    ADom.nnzDom = {0..#indices.size};
+    ADom.idx = indices;
+
+    return ADom;
+  }
+
+  /* Warning
+     This method has been deprecated.  Please use
+       CSRMatrix(parentDom: domain(2), data: [?nnzDom] ?eltType, indices: [nnzDom], indptr: [?indDom])
+     instead.
+
+     Return a CSR matrix constructed from internal representation:
+    - ``shape``: bounding box dimensions
+    - ``data``: non-zero element values
+    - ``indices``: non-zero row pointers
+    - ``indptr``: index pointers
+  */
   proc CSRMatrix(shape: 2*int, data: [?nnzDom] ?eltType, indices: [nnzDom], indptr: [?indDom])
     where indDom.rank == 1 && nnzDom.rank == 1 {
+    halt("CSRMatrix only given the matrix extents is deprecated.  Please use the CSRMatrix that takes a domain(2) as first parameter instead.");
     var ADom = CSRDomain(shape, indices, indptr);
     var A: [ADom] eltType;
     A.data = data;
     return A;
   }
 
-  /* Return a CSR domain constructed from internal representation */
+  /* Warning
+     This method has been deprecated.  Please use
+       CSRDomain(parentDom: domain(2), indices: [?nnzDom], indptr: [?indDom])
+     instead.
+
+     Return a CSR domain constructed from internal representation */
```

Comment:
I would suggest the following format for our deprecation warning in this functionn and the `CSRMatrix` one:

```chpl
/* Return a CSR domain constructed from internal representation.

.. warning:: 
     This function has been deprecated.  
     Please use ``CSRDomain(parentDom: domain(2), indices: [?nnzDom], indptr: [?indDom])`` instead.
 ```

This will render as a red box with the deprecation warning.

Reply:
Good to know.  Should we update https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/bestPractices/Deprecation.rst to reflect this?

Reply:
I think so